### PR TITLE
Doc/#746 tutorials

### PR DESCRIPTION
**Pull request by jrobinAV**
### What I Did:
- Removed the "Getting-started" folder and the Python Step that was generating it.
- Renamed the "Getting-started-new" folder to the "Getting-started" folder.
- Created a "Tutorials" folder with an index page listing all our tutorials.
- I moved the 3 old GS into the "tutorials" folder and renamed them. Each one is in a dedicated subfolder.

image

### What you need to review:
- the structure above with the files `mkdocs.yml_template` and `docs/tutorials/index.md`.
- the removal of the Getting Started generation step with the files `tools/_setup_generation/setup.py` and `tools/_setup_generation/step_getting_started.py`
All other changes are some files moved or renamed.

### Remaining work:
1. Review and update the list of tutorials to include. We may need to change, merge or remove some tutorials since we are adding the 3 old Getting Started.

2. Replace the tutorials/index.md page. I want a list of all our tutorials. It includes the 3 old Getting Started and our current tutorials on the website. I would like to turn the table into a beautiful page with a similar layout to the one on the website but without the Core/Gui split:
TMP_PIC_TO_REMOVE

3. Create the new layout with beautiful CSS to apply to the tutorials/index.md page.

4. Review the tutorials individually to improve the content, the style, and the consistency.

5. Archive the 3 old GS repositories on GitHub.

### CODE REVIEW

File: docs/tutorials/understanding_gui/step_02/ReadMe.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+Taipy GUI can be considered as an **augmented** Markdown; it adds the concept of 
+**[Visual elements](https://docs.taipy.io/en/latest/manuals/gui/viselements/)** on top of all the Markdown syntax. A visual 
+element is a Taipy graphical object displayed on the client. It can be a 
+[slider](https://docs.taipy.io/en/latest/manuals/gui/viselements/slider/), a 
+[chart](https://docs.taipy.io/en/latest/manuals/gui/viselements/chart/), a 
+[table](https://docs.taipy.io/en/latest/manuals/gui/viselements/table/), an 
+[input](https://docs.taipy.io/en/latest/manuals/gui/viselements/input/), a 
+[menu](https://docs.taipy.io/en/latest/manuals/gui/viselements/menu/), etc. Check the list 
+[here](https://docs.taipy.io/en/latest/manuals/gui/controls/).
```
Comment (FlorianJacta) [inline docs/tutorials/understanding_gui/step_02/ReadMe.md:22]: Broken link here: https://docs.taipy.io/en/latest/manuals/gui/viselements/controls/

**Comment by FlorianJacta** [comment]
The Getting Started from complete_application is the old Getting Started

**Comment by FlorianJacta** [comment]
Tutorials on the website are YouTube videos. What we would want to transfer are more [tips](https://www.taipy.io/tips/) of taipy.io?

File: docs/getting_started/index.md [hunk 2]
```diff
@@ -1,15 +1,171 @@
...
+
+Include the following code in your Python script:
+
+```python linenums="1"
+from taipy import Core
+
+if __name__ == "__main__":
+    Core().run()
+```
+Line 3 is standard boilerplate code to ensure your script runs only from the main module,
```
Comment (erikfTW) [inline docs/getting_started/index.md]:
```suggestion
- Line 3 is standard boilerplate code to ensure your script runs only from the main module, 
```

File: docs/getting_started/index.md [hunk 3]
```diff
@@ -1,15 +1,171 @@
...
+```python linenums="1"
+from taipy import Core
+
+if __name__ == "__main__":
+    Core().run()
+```
+Line 3 is standard boilerplate code to ensure your script runs only from the main module, 
+preventing accidental execution. It is strongly recommended to include it.
+
+In line 4, we instantiate and run a Core service.
```
Comment (erikfTW) [inline docs/getting_started/index.md]:
```suggestion
- Line 4 instantiates and runs a Core service.
```

File: docs/getting_started/index.md [hunk 4]
```diff
@@ -1,15 +1,171 @@
...
+hello_scenario = tp.create_scenario(scenario_cfg)
+hello_scenario.input_name.write("Taipy")
+hello_scenario.submit()
+print(hello_scenario.message.read())
+```
+
+In line 3, method `tp.create_scenario()` instantiates the new scenario name *hello_scenario* 
+from the scenario configuration built before.
+
+Line 4, sets the input data node *input_name* of *hello_scenario* with the string value "Taipy"
```
Comment (erikfTW) [inline docs/getting_started/index.md]:
```suggestion
- Line 4, sets the input data node *input_name* of *hello_scenario* with the string value "Taipy" 
```

File: docs/getting_started/index.md [hunk 5]
```diff
@@ -1,15 +1,171 @@
...
+print(hello_scenario.message.read())
+```
+
+In line 3, method `tp.create_scenario()` instantiates the new scenario name *hello_scenario* 
+from the scenario configuration built before.
+
+Line 4, sets the input data node *input_name* of *hello_scenario* with the string value "Taipy" 
+using the `write()` method.
+
+Line 5 submits the *hello_scenario* for execution, which triggers the creation and execution of
```
Comment (erikfTW) [inline docs/getting_started/index.md]:
```suggestion
- Line 5 submits the *hello_scenario* for execution, which triggers the creation and execution of 
```

File: docs/getting_started/index.md [hunk 6]
```diff
@@ -1,15 +1,171 @@
...
+from the scenario configuration built before.
+
+Line 4, sets the input data node *input_name* of *hello_scenario* with the string value "Taipy" 
+using the `write()` method.
+
+Line 5 submits the *hello_scenario* for execution, which triggers the creation and execution of 
+a job. This job reads the input data node, passes the value to the *build_message()* function, 
+and writes the result to the output data node.
+
+Line 6 reads and prints the output data node *message* written by the execution of the scenario
```
Comment (erikfTW) [inline docs/getting_started/index.md]:
```suggestion
- Line 6 reads and prints the output data node *message* written by the execution of the scenario 
```

File: docs/getting_started/index.md [hunk 7]
```diff
@@ -1,15 +1,171 @@
...
+
+This involves a single input data node called *name*. Afterward, a task called *build_message* 
+takes the first initial data node and returns a second data node called *message*.
+
+Building the corresponding Taipy application requires three easy steps.
+
+### Configuration
+
+The first step involves configuring the execution graph. This includes 
+configuring data nodes, tasks, and scenario itself.
```
Comment (erikfTW) [inline docs/getting_started/index.md]:
```suggestion
The initial step is to set up the execution graph, which involves configuring data nodes, tasks, and the scenario itself. 
```

**Review by erikfTW — CHANGES_REQUESTED** [review]

File: docs/tutorials/understanding_gui/step_02/ReadMe.md (hunk 1 above)
Comment (jrobinAV) [reply docs/tutorials/understanding_gui/step_02/ReadMe.md:22]: Thx. I did not change the content of the old GS at all, and all the links must be updated anyway (we must have relative paths and not absolute paths anymore)
I propose not to fix the link in this PR.
We need to synchronize on how to turn the Getting Started into a tutorial. The content should be changed slightly...

File: docs/getting_started/index.md [hunk 8]
```diff
@@ -1,15 +1,171 @@
...
+
+![result](result.png){width=50%}
+
+Combining both the scenario and the Gui parts, you can download the entire Python code for the 
+example from the link:
+<a href="./hello_world.py" download>`hello_world.py`</a>
+
+That covers the fundamental steps to begin with Taipy. For more realistic 
+use cases and advanced features, refer to the 
+[Tutorials](../tutorials/index.md), [demos](../demos/index.md), or
```
Comment (jrobinAV) [inline docs/getting_started/index.md]:
```suggestion
[tutorials](../tutorials/index.md), [demos](../demos/index.md), or 
```

**Comment by jrobinAV** [comment]
> Tutorials on the website are YouTube videos. What we would want to transfer are more [tips](https://www.taipy.io/tips/) of taipy.io?

Both, actually. I think this is discussable. The purpose of this PR is to review the structure. We can discuss what will be the content tomorrow.

**Pull request merged**